src/pages/top-up/Swap.tsx

```diff
@@ -52,28 +47,65 @@ export function Swap({ header }: Props): ReactElement {
   const navigate = useNavigate()
   const { enqueueSnackbar } = useSnackbar()
 
+  // Fetch current price of BZZ
   useEffect(() => {
     // eslint-disable-next-line no-console
     getBzzPriceAsDai(desktopUrl).then(setPrice).catch(console.error)
   }, [desktopUrl])
 
-  if (!balance || !nodeAddresses) {
-    return <Loading />
-  }
+  // Set the initial xDAI to swap
+  useEffect(() => {
+    if (!balance) {
+      return
+    }
 
-  const optimalSwap = balance.dai.minusBaseUnits('1')
-  const lowAmountSwap = new DaiToken(balance.dai.toBigNumber.dividedToIntegerBy(2))
+    if (balance.dai.toDecimal.isGreaterThanOrEqualTo(DaiToken.fromDecimal('1').plusBaseUnits(MINIMUM_XDAI).toDecimal)) {
```

Comment:
Can you please move out the `DaiToken.fromDecimal('1').plusBaseUnits(MINIMUM_XDAI).toDecimal)` into a separate variable for better readability?